### DISCUSSION TITLE
Git add and commit

### DISCUSSION THREAD

**Discussion by Peterkenya** [community/community#59720] [New to GitHub]
Hi iam facing serious challenge with my Github password,i cant commit my codes from the sandbox to the github code space

**Comment by qjoly** [comment]
Hi,

Are you aware that it's no longer possible to use your password during a `git push` in HTTPS? 

Your only option is to use the SSH protocol (by setting a private key in your Github settings) or to generate a token that will replace your password when using the https protocol.

Here's a step-by-step tutorial for generating a token: https://medium.com/@obielum1858/using-personal-access-tokens-with-git-and-github-to-push-4fd22ef79db8

**Comment by Gustavonn07** [comment]
I'm sorry to hear that you're having trouble with your GitHub password. Here are a few steps you can take to troubleshoot the issue:

1. Reset your password: Go to the GitHub login page and click on the "Forgot password?" link. Follow the instructions provided to reset your password. Make sure to choose a strong and secure password.

2. Clear your browser cache: Sometimes, cached data can interfere with the login process. Clear your browser cache and try logging in again.

3. Check for any GitHub service disruptions: Occasionally, GitHub may experience service disruptions or outages. Visit the GitHub status page (https://www.githubstatus.com/) to check if there are any ongoing issues that might be affecting your ability to commit code.

4. Verify your credentials: Double-check that you're using the correct username and password combination. Passwords are case-sensitive, so ensure that you're entering the characters correctly.

5. Use personal access tokens (PAT): Instead of using your account password, you can create a personal access token and use it as your password for authentication. Personal access tokens provide a more secure way to authenticate with GitHub. To create a personal access token, go to your GitHub account settings, navigate to the "Developer settings" section, and then click on "Personal access tokens."

If none of these steps resolve the issue, you may want to reach out to GitHub support directly for further assistance. They can provide more specific guidance based on your account and the issue you're facing.

If your problem is with configure Git credentials: Run the commands below in the VS Code terminal to configure your Git username and email address. Replace "Your Name" with your name and "your-email@example.com" with your GitHub email address:

1. `git config --global user.name "Your Name"`
    `git config --global user.email "your-email@example.com"`

2. Connect to GitHub: Now, you need to add your GitHub credentials to connect to your remote repository. You can use HTTPS or SSH for authentication. If you want to use HTTPS, just clone the repository or add the remote repository manually. To clone, go to the VS Code top menu, choose "View" and select "Command Palette" or press "Ctrl+Shift+P" (or "Cmd+Shift+P" on macOS) . In the command palette, type "Git: Clone" and follow the instructions to clone your repository.

3. Get started with Git: With Git set up and connected to your GitHub repository, you can start using Git commands to commit, push, and pull your changes. You can also use Git-related VS Code extensions like GitLens for a more advanced experience.

(Commands):
`git init`
`git checkout -b (branch name)`
`git add (file) (or .)`
`git commit -m "coment"`
`git push origin (branch name)`

Hope it helps!!

**Comment by Peterkenya** [comment]
Hello Community/Community Support,

I urgently need help, I happen to have  enabled double security
authentication on my github account unfortunately have misplaced the
recovery codes and don't have a mobile app. kindly disable for me because i
have many codes to run in the github repository but cant get in please.

Thanks,
Peter
